src/link.d

```diff
@@ -773,21 +808,40 @@ version (Windows)
         cmd = toWinPath(cmd);
         version (CRuntime_Microsoft)
         {
-            if (strchr(cmd, ' '))
+            // Open scope so dmd doesn't complain about alloca + exception handling
             {
-                // MSVCRT: spawn does not work with spaces in the executable
-                size_t cmdlen = strlen(cmd);
-                char* shortName = (new char[](cmdlen + 1)).ptr; // enough space
-                uint plen = GetShortPathNameA(cmd, shortName, cast(uint)cmdlen + 1);
-                if (plen > 0 && plen <= cmdlen)
-                    cmd = shortName;
+                // Use process spawning through the WinAPI to avoid issues with executearg0 and spawnlp
+                OutBuffer cmdbuf;
+                cmdbuf.writestring("\"");
+                cmdbuf.writestring(cmd);
+                cmdbuf.writestring("\" ");
+                cmdbuf.writestring(args);
+
+                STARTUPINFOA startInf;
```

Comment:
I changed the code to forward the std handles to the newly created process. I also tested that it works.